### DISCUSSION TITLE
Hide Navbar and Footer

### DISCUSSION THREAD

**Discussion by naanand** [facebook/docusaurus#5952] [Q&A]
I have added a react component to include a drawer in my docs. But the drawer opens with the default navbar and footer. How can I hide the navbar and footer? Is anything like this feasible? Any alternatives?

`--- 
navbar: false
hide_title: true
hide_table_of_contents: true
footer: false
---`

**Comment by Josh-Cena** [comment]
The doc page doesn't control the site's layout. Could you give a screenshot of what's the problem like? If you make a custom React page you can hide the navbar by not using the `Layout` comp, but it's impossible with doc pages.

**Reply by naanand** [reply]
<img width="1440" alt="Screen Shot 2021-11-15 at 4 28 27 PM" src="https://user-images.githubusercontent.com/84015140/141770396-e8afdf1d-e6f2-4362-8f10-fb4acd16286c.png">

**Reply by Josh-Cena** [reply]
That's interesting... How did you set this component up? Is there code that I can inspect?

**Reply by naanand** [reply]
Disclaimer: I have no coding background. So I have just used the docusaurus documentation and other links from Google to get to this stage. 

This is the main markdown file. I import ./networks.table here. This is a .js file.

```md
--- 
sidebar_position: 1
hide_table_of_contents: true
---

# Network Guides

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import PesterDataTable from '@site/src/components/PesterDataTable';
import { sdkattr, s2sattr, s2snetworks, sdknetworks } from './networks.table';
```

I have these dependencies in network.table.js
```jsx
import { Observe, State } from 'mdx-observable';

import { Drawer } from 'rsuite';
import MesonMDX from './MesonMDX';
import React from "react";
```

This is the code in network.table.js.

```jsx
  Cell: ({ cell: { value }, row: { original } }) => (
    <State initialState={{
      isDrawerOpen: false
    }}>            

      <Observe>
      {({isDrawerOpen, setState}) => {  
        return (          
        <div align="center">
          <SettingOutlined onClick={e => {
            setState(s => {       
              return {isDrawerOpen : true}
            });
          }}/>
        </div>
      )}}
      </Observe>
      
      <Observe>
      {({isDrawerOpen, setState}) => {   
      return (
        <div align="center">
          <Drawer
            open={isDrawerOpen}
            size="md"
            onClose={e => {
              setState(s => {
                return {isDrawerOpen : false}
              });
            }}
          >
            <Drawer.Header>
              <Drawer.Title><div><h1 style={{ padding: 0, margin: 0, align: 'end' }}>Setup {original.name}</h1></div></Drawer.Title>
            </Drawer.Header>
            <Drawer.Body>
              <MesonMDX fileName={original.guide} />
            </Drawer.Body>
          </Drawer>                  
        </div>
      )}}        
      </Observe>

    </State>
    ),
```

I ran into a window not defined error while building the project and hence wrote the following to overcome that. 

```jsx
import React from "react"
const Placeholder = () => null
const MesonMDX = props => {

  const [Component, setComponent] = React.useState(() => Placeholder)

  React.useEffect(() => {
    import("react-md-file").then(Thing => setComponent(() => Thing.default))
  }, [])
  return <Component {...props} />
}
export default MesonMDX
```

**Reply by slorber** [reply]
It will be hard to help you without acceding to your source code 😅 

![image](https://user-images.githubusercontent.com/749374/142038282-3837d172-fa96-41b4-93ca-997397180557.png)

This is not Docusaurus rendering this drawer and any of it's content, so we can't know how it ends-up there

--- 


> I ran into a window not defined error while building the project and hence wrote the following to overcome that.

We have a component for that: https://docusaurus.io/docs/docusaurus-core#browseronly

**Reply by lbochsler** [reply]
Most likely, one of your imported components (e.g. @site/src/components/PesterDataTable) returns a React `<Layout>` section. Just remove the Layout part and the nav-bar will disappear.
